viewing_party/party.py

```diff
@@ -1,23 +1,152 @@
 # ------------- WAVE 1 --------------------
 
+#from tkinter.tix import InputOnly
+
 def create_movie(title, genre, rating):
-    pass
 
-# -----------------------------------------
+    new_movie = {}
+    if title and genre and rating:
+        new_movie["title"] = title
+        new_movie["genre"]= genre
+        new_movie["rating"]= rating
+        return new_movie
+    else:
+        return None
+
+def add_to_watched(user_data, movie):
+    watched_movies = []
+
+    user_data.append(movie)
+
+    return user_data
+
+def add_to_watched_list(user_data, movie):
+    watchlist = []
+
+    user_data.append(movie)
+
+    return user_data
+
+def watch_movie(user_data, title):
+
+    for items in user_data["watchlist"]:
+        if title in user_data["watchlist"]:
+            user_data["watched"].append(user_data["watchlist"][0])
+            user_data["watchlist"].pop(0)
+        return user_data

```

Comment:
There's quite a few issues here.

`for items in user_data["watchlist"]` will go through each element of the `user_data["watchlist"]` list setting it to `items`. However, the code does not actually refer to `items` so this will just needlessly run the loop one time for each item in the list.

The `title` parameter that gets passed in will be a string. However, `user_data["watchlist"]` is a list of dictionaries. Thus, `title in user_data["watchlist"]` will never be `True`, and the code inside the block will never run.

Even if the code in the `if` block had run, it would not do the right thing. Rather than making sure to use `title` to find the right index within `user_data["watchlist"]`, the code is directly appending and popping the zeroth element  of the list. However, that may not actually be the movie in the list with that title. 

On that last issue, our tests would not have caught it, and admittedly that is on us. However, the important point is that by the description of the problem, we can't make the assumption that our movie to remove from `user_data["watchlist"]` will be the zeroth index, since there's nothing in particular forcing the movie with `"title"` key `title` to be that one.